File: api/schemas/hldtech.py
from pydantic import BaseModel, UUID4
from typing import Optional
from datetime import datetime

class HLDTechBase(BaseModel):
    id: UUID4
    project_id: UUID4
    original_name: str
    stored_name: str
    file_path: str
    created_at: datetime

    class Config:
        from_attributes = True

class HLDTechCreate(BaseModel):
    project_id: UUID4
    original_name: str
    stored_name: str
    file_path: str

    class Config:
        from_attributes = True

class HLDTechUpdate(BaseModel):
    original_name: Optional[str] = None
    

    class Config:
        from_attributes = True

File: client/schema/hldtech.ts
export interface HLDTech {
    id: string;
    project_id: string;
    original_name: string;
    stored_name: string;
    file_path: string;
    created_at: Date;
}
    

export interface HLDTechCreate {
    project_id: string;
    original_name: string;
    stored_name: string;
    file_path: string;
}

export interface HLDTechUpdate {
    original_name: string;
}